analyzers/src/SonarAnalyzer.CSharp/Extensions/SonarAnalysisContextExtensions.cs

```diff
@@ -46,7 +46,7 @@ public static class SonarAnalysisContextExtensions
     public static void RegisterCodeBlockStartAction(this SonarAnalysisContext context, Action<SonarCodeBlockStartAnalysisContext<SyntaxKind>> action) =>
         context.RegisterCodeBlockStartAction(CSharpGeneratedCodeRecognizer.Instance, action);
 
-    [Obsolete("Use overload without Diagnostic.Create - the same as in the context class.")]
+    [Obsolete("Use another overload of ReportIssue, without calling Diagnostic.Create")]
     public static void ReportIssue(this SonarCompilationReportingContext context, Diagnostic diagnostic) =>
```

Comment:
Can the obsolete methods be removed now?

Reply:
Discussed on Slack.

Reply:
Yes for the extensions - done in #9356 
No for the base classes, there are too many usages).